x-pack/plugins/apm/server/lib/services/get_service_transaction_groups/index.ts

```diff
@@ -0,0 +1,322 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { ValuesType } from 'utility-types';
+import { orderBy } from 'lodash';
+import {
+  TRANSACTION_PAGE_LOAD,
+  TRANSACTION_REQUEST,
+} from '../../../../common/transaction_types';
+import { EventOutcome } from '../../../../common/event_outcome';
+import { PromiseReturnType } from '../../../../../observability/typings/common';
+import { rangeFilter } from '../../../../common/utils/range_filter';
+import {
+  EVENT_OUTCOME,
+  SERVICE_NAME,
+  TRANSACTION_NAME,
+  TRANSACTION_TYPE,
+} from '../../../../common/elasticsearch_fieldnames';
+import { Setup, SetupTimeRange } from '../../helpers/setup_request';
+import { getBucketSize } from '../../helpers/get_bucket_size';
+import {
+  getProcessorEventForAggregatedTransactions,
+  getTransactionDurationFieldForAggregatedTransactions,
+} from '../../helpers/aggregated_transactions';
+
+export type ServiceTransactionGroupItem = ValuesType<
+  PromiseReturnType<typeof getServiceTransactionGroups>['transaction_groups']
+>;
```

Comment:
I think we should prefer to use the `APIReturnType` helper over `PromiseReturnType` since the former is the true type of the api response.